src/truncate.jl

```diff
@@ -26,10 +26,17 @@ function truncated(d::UnivariateDistribution, l::T, u::T) where {T <: Real}
     else
         logcdf(d, l)
     end
+    _T = typeof(loglcdf)
+    if isfinite(loglcdf) && l == -Inf
+        loglcdf = convert(_T, -Inf)
```

Comment:
Can you explain why this fixes the problem? It shouldn't change the value of `loglcdf`?

Reply:
It doesn't change its value but it changes the partial to 0 from NaN which gives the correct behaviour.
```julia
julia> using ForwardDiff

julia> d = ForwardDiff.Dual(1.0, 1.0)
Dual{Nothing}(1.0,1.0)

julia> convert(typeof(d), Inf)
Dual{Nothing}(Inf,0.0)
```